packages/nouns-contracts/test/foundry/NounsAuctionHouseV3.t.sol

```diff
@@ -957,24 +973,68 @@ contract NounsAuctionHouseV2_setPricesTest is NoracleBaseTest {
     }
 }
 
-contract NounsAuctionHouseV2_OwnerFunctionsTest is NounsAuctionHouseV2TestBase {
+contract NounsAuctionHouseV2_OwnerFunctionsTest is NounsAuctionHouseV3TestBase {
     function test_setTimeBuffer_revertsForNonOwner() public {
         vm.expectRevert('Ownable: caller is not the owner');
         auction.setTimeBuffer(1 days);
     }
 
     function test_setTimeBuffer_revertsGivenValueAboveMax() public {
-        vm.prank(auction.owner());
+        vm.prank(IOwner(address(auction)).owner());
         vm.expectRevert('timeBuffer too large');
         auction.setTimeBuffer(1 days + 1);
     }
 
     function test_setTimeBuffer_worksForOwner() public {
         assertEq(auction.timeBuffer(), 5 minutes);
 
-        vm.prank(auction.owner());
+        vm.prank(IOwner(address(auction)).owner());
         auction.setTimeBuffer(1 days);
 
         assertEq(auction.timeBuffer(), 1 days);
     }
+
+    function test_setStreamEscrowParams_revertsForNonOwner() public {
+        vm.expectRevert('Ownable: caller is not the owner');
+        auction.setStreamEscrowParams(1, 2, address(3));
+    }
+
+    function test_setStreamEscrowParams_worksForOWner() public {
+        vm.prank(IOwner(address(auction)).owner());
+        vm.expectEmit();
+        emit IAH.StreamEscrowParamsUpdated(1000, 500, address(123));
+        auction.setStreamEscrowParams({
+            _immediateTreasuryBPs: 1000,
+            _streamLengthInTicks: 500,
+            _streamEscrow: address(123)
+        });
+        assertEq(auction.immediateTreasuryBPs(), 1000);
+        assertEq(auction.streamLengthInTicks(), 500);
+        assertEq(address(auction.streamEscrow()), address(123));
+    }
+
+    function test_setImmediateTreasuryBPs_revertsForNonOwner() public {
```

Comment:
for completeness worth adding the positive case of this function, inc. checking it emits?

Reply:
sure bb71819a5b2eef22f787847d7d627ce3b5e82977